plugin/src/main/resources/io/jenkins/plugins/analysis/warnings/groovy/ParserConfiguration/help-consoleLogScanningPermitted.html

```diff
@@ -0,0 +1,14 @@
+<div>
+  <p>
+    Console Log parsing has to be processed directly on the Jenkins controller
+    which means running the parser groovy code (configured here) directly on the controller which <em>CAN BE UNSAFE</em> from a security point of view.
+  </p>
+    e.g. if your groovy code is poorly written.
+  <p>
+    If you do not accept this risk then you should not enable this feature
+    ...
+    but if you do not enable this feature then the groovy parsers that you define will only be able to parse files in the build workspace
+    (safely processed on the agents responsible for the builds rather than natively on the controller)
+    rather than the build console log itself.
+  </p>
```

Comment:
This text should be reviewed by someone who's 100% aware of the Jenkins-security-team's concerns about this.

Reply:
@daniel-beck @Wadeck Is this ok for you?

Reply:
As I wrote elsewhere, I think we need to revisit the security considerations for Overall/Administer users after we retired the RunScripts permission around Jenkins 2.222.

That said, I recommend you consider whether you want to allow users with Overall/Manage permission (JEP-223) to configure these parsers at some point in the near future.

FWIW the real long term solution for all of this seems to be Script Security integration anyway. Then none of this is a concern -- an admin will not be impacted (their scripts will be pre-approved), Overall/Manage users' parsers need to be approved by admins first or sandboxes, and the scripts can run wherever.

Reply:
Yes  I view this PR as only being useful until this plug-in has sandboxed scripts.
(and if I hadn't been in a big hurry to "just get something working now!" then I would've looked into that instead).
Once scripts are sandboxed then this configuration option is irrelevant and can be hard coded as "on".

Reply:
I implemented the script security integration already some years ago. I removed that part because then the parsers did not work on the agent anymore. Script security is only available on the controller side.